redis/src/main/scala/zio/redis/RespArgument.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 John A. De Goes and the ZIO contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.redis
+
+import zio.Chunk
+import zio.redis.RespValue.BulkString
+import zio.schema.Schema
+import zio.schema.codec.BinaryCodec
+
+import java.nio.charset.StandardCharsets
+
+sealed trait RespArgument {
+  def respValue: RespValue.BulkString
+}
+
+object RespArgument {
+
+  final case class Unknown(bytes: Chunk[Byte]) extends RespArgument {
+    lazy val respValue: BulkString = RespValue.BulkString(bytes)
+  }
+
+  object Unknown {
+    def apply(str: String): Unknown                                            = Unknown(Chunk.fromArray(str.getBytes(StandardCharsets.UTF_8)))
+    def apply[A](data: A)(implicit codec: BinaryCodec, schema: Schema[A]): Key = Key(codec.encode(schema)(data))
+  }
+
+  final case class CommandName(str: String) extends RespArgument {
+    lazy val respValue: BulkString = RespValue.bulkString(str)
+  }
+
+  final case class Literal(str: String) extends RespArgument {
+    lazy val respValue: BulkString = RespValue.bulkString(str)
+  }
```

Comment:
Just to clarify, `Value` is the generic value holder (i.e., anything that can be encoded as chunk of bytes) while `Literal` is to be used mostly internally wherever we're dealing with string literals (e.g. command options) ?

Reply:
Yeah, But may be I thought about `Value` more like something that we want to save in redis and that's why there are we have `Unknown` case - where i don't know is it have to have some separate case for example like group-name or consumer or it should be a `Value` like ip address or ttl duration 🤔